utils/bin/poller/src/main.rs

```diff
@@ -0,0 +1,301 @@
+                            &*response.returnData[pre_interactions.len()].returnData,
+                            true
+                        )?,
+                        IERC1271::isValidSignatureCall::SELECTOR,
+                        "Signature was not valid"
+                    );
+
+                    // Check that all post interactions were successful
+                    response.returnData[pre_interactions.len() + 1..]
+                        .iter()
+                        .for_each(|r| {
+                            assert!(r.success, "Post interaction Call failed");
+                        });
+
+                    // If legacy, check there is another interaction that returns the commitment
+                    // after the post interaction is meant to set it back to `bytes32(0)`
+                    if legacy {
+                        assert_eq!(
+                            response.returnData.len(),
+                            length + 1,
+                            "Missing expected commitment check"
+                        );
+                        assert!(
+                            response.returnData[length].success,
+                            "Commitment check failed"
+                        );
+                        assert_eq!(
+                            **response.returnData[length].returnData,
+                            FixedBytes::<32>::default(),
+                            "Commitment was not reset"
+                        );
+                    }
+
+                    println!("\nGenerated order was able to be settled successfully!");
+                }
+                Err(e) => {
+                    println!("Error: {:?}", e);
+                }
+            }
+        }
+        Err(Error::TransportError(TransportError::ErrorResp(err))) => {
+            let data = err.data.unwrap_or_default();
+            let data = data.get().trim_matches('"');
+            let data = Bytes::from_str(data).unwrap();
+            println!("Data: {:?}", data);
+            let decoded_error = ConstantProductHelperErrors::abi_decode(&data, true);
+
+            match decoded_error {
+                Ok(t) => match t {
+                    ConstantProductHelperErrors::PoolDoesNotExist(_) => {
+                        println!("Pool does not exist: {:?}", amm);
+                    }
+                    ConstantProductHelperErrors::PoolIsClosed(_) => {
+                        println!("Pool is closed: {:?}", amm);
+                    }
+                    _ => {}
+                },
+                Err(e) => match decode_revert_reason(&data) {
+                    Some(reason) => {
+                        println!("Reason: {:?}", reason);
+                    }
+                    None => {
+                        println!("Err: {:?}", e);
+                    }
+                },
+            }
+        }
+        Err(e) => {
+            println!("Error: {:?}", e);
+        }
```

Comment:
I think there's too many `match` branches, here is a stack of three. Context is easily lost, I had no clue what this is matching for. I'd say that for things like this we should just check if the variable is an error, if yes stop in some way (return Err?) otherwise continue with the unwrapped variable.

Reply:
I'd generally favour the use of `match` to enforce exhaustive processing so that there aren't any possible code paths that haven't been considered. Sure, there's a lot of matches here, but also this is an unfortunate by-product of integrating legacy and non-legacy (already ensuring at least 1 branch)